### PR TITLE
Refactor metrics

### PR DESCRIPTION
**Pull request by nellshamrell**
Small refactor that better lays the foundation for adding more quality metrics.

### CODE REVIEW

**Comment by nellshamrell** [comment]
Confirmed that metrics still work as expected when uploading cookbooks locally

File: src/fieri/spec/models/collaborator_worker_spec.rb [hunk 1]
```diff
@@ -33,8 +33,9 @@
     CollaboratorWorker.new.perform(cookbook_name)
 
     assert_requested(:post, "#{ENV['FIERI_SUPERMARKET_ENDPOINT']}/api/v1/cookbook-versions/collaborators_evaluation", times: 1) do |req|
-      req.body =~ /collaborator_failure=true/
-      req.body =~ /collaborator_feedback=.+/
+      expect(req.body).to include("cookbook_name=#{cookbook_name}")
+      expect(req.body).to include('collaborator_failure=false')
+      expect(req.body).to include('collaborator_feedback=passed')
```
Comment (robbkidd) [inline src/fieri/spec/models/collaborator_worker_spec.rb]: Yay assertions! 👍

File: src/supermarket/app/controllers/api/v1/cookbook_versions_controller.rb [hunk 2]
```diff
@@ -1,4 +1,7 @@
 class Api::V1::CookbookVersionsController < Api::V1Controller
+  before_action :check_cookbook_name_present, only: [:foodcritic_evaluation, :collaborators_evaluation]
+  before_action :check_authorization, only: [:foodcritic_evaluation, :collaborators_evaluation]
+  before_action :find_cookbook_version, only: [:foodcritic_evaluation, :collaborators_evaluation]
```
Comment (robbkidd) [inline src/supermarket/app/controllers/api/v1/cookbook_versions_controller.rb]: Nice rollup. Sets things up nicely for more refactoring (mild renovation?) in the future for what will be a growing number of evaluation endpoints.

Comment (nellshamrell) [inline src/supermarket/app/controllers/api/v1/cookbook_versions_controller.rb]: 👍

**Pull request merged**